### PR TITLE
add docs for revoke token by id

### PR DESCRIPTION
**Pull request by tomsik68**

### CODE REVIEW

File: docs/workflows/api-token.md [hunk 1]
```diff
@@ -95,6 +95,11 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
 
 ## Revoking Access Tokens
 
+Depending on what information is available to you, you may choose to use one of the ways to revoke an access token.
+It's recommended to use RFC7009 revoke endpoint if you can.
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:99]: Why is this recommended over other options? Maybe a quick sentence explaining that or the main difference?

File: docs/workflows/api-token.md [hunk 2]
```diff
@@ -112,6 +117,23 @@ curl='curl "https://auth-$(REGION).beyondidentity.com/v1/tenants/$(TENANT_ID)/re
...
+authentication containing a management API token with the `tokens:delete`
+scope.
+
+You must also know the `TOKEN_ID` of the token you want to revoke. This can be
+returned for example from the token listing.
+
+<MultiLanguageCodeBlock
+curl='curl "https://api-$(REGION).beyondidentity.com/v1/tenants/$(TENANT_ID)/realms/$(REALM_ID)/tokens/$(TOKEN_ID)" \
+-X DELETE \
+-H "Authorization: Bearer $(MANAGEMENT_API_TOKEN)" \
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:132]: Should the above curl in `RFC7009 revoke endpoint` also use `MANAGEMENT_API_TOKEN` too?

File: docs/workflows/api-token.md [hunk 3]
```diff
@@ -95,6 +95,11 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
 
 ## Revoking Access Tokens
 
+Depending on what information is available to you, you may choose to use one of the ways to revoke an access token.
+It's recommended to use RFC7009 revoke endpoint if you can.
+
+### RFC7009 revoke endpoint
+
 In order to revoke an access token, you must be passed authentication in the form of either Bearer or Basic.
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:103]: Does this option also "revoke a token via management API"?

File: docs/workflows/api-token.md (hunk 1 above)
Comment (tomsik68) [reply docs/workflows/api-token.md]: You're right, this is not a good explanation. I updated the documentation to say "if you have raw token, use RFC-7009 endpoint. if you have token id, use management api". Would this make more sense?

File: docs/workflows/api-token.md (hunk 3 above)
Comment (tomsik68) [reply docs/workflows/api-token.md:104]: Yes, it does use management API under the hood. The outcome of these two endpoints is the same, it's just a different API.

File: docs/workflows/api-token.md (hunk 2 above)
Comment (tomsik68) [reply docs/workflows/api-token.md:133]: Great catch, this is also a management API token! Plus, we also offer Basic auth version of this endpoint so I'll try adding that too.

File: docs/workflows/api-token.md [hunk 4]
```diff
@@ -95,6 +95,12 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
...
 
+Depending on what information is available to you, you may choose to use one of the ways to revoke an access token.
+
+If you have the raw access token, use the RFC-7009 revoke endpoint. If you only have the token ID (for example from active token listing), you can also revoke the token by token_id.
+
+### RFC7009 revoke endpoint
+
 In order to revoke an access token, you must be passed authentication in the form of either Bearer or Basic.
 
 In the case of Basic authentication, the passed authentication must come from the confidential Application that the token has been minted for. The passed access token must be signed by the same client id as the application.
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:106]:
```suggestion
In the case of Basic authentication, the passed authentication must come from the confidential application that the token has been minted for. The passed access token must be signed by the same client id as the application.
```

File: docs/workflows/api-token.md (hunk 3 above)
Comment (turnipdabeets) [reply docs/workflows/api-token.md:104]: Down below for `### Revoking token by token_id` says:

"In order to revoke a token via management API, you must pass a Bearer
 authentication containing a management API token with the tokens:delete
 scope." 
 
Do you think we should should mention the management API here too? Or if not, I'm trying to figure what you mean by the management API below? Is the /token part of management API and /revoke part of something else?

File: docs/workflows/api-token.md [hunk 5]
```diff
@@ -95,6 +95,12 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
 
 ## Revoking Access Tokens
 
+Depending on what information is available to you, you may choose to use one of the ways to revoke an access token.
+
+If you have the raw access token, use the RFC-7009 revoke endpoint. If you only have the token ID (for example from active token listing), you can also revoke the token by token_id.
+
+### RFC7009 revoke endpoint
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:102]:
```suggestion
### RFC7009 Revoke Endpoint
```

File: docs/workflows/api-token.md [hunk 6]
```diff
@@ -106,12 +112,29 @@ Note that passing an invalid token, or a token which has already been revoked or
...
 curl='curl "https://auth-$(REGION).beyondidentity.com/v1/tenants/$(TENANT_ID)/realms/$(REALM_ID)/applications/$(MANAGEMENT_APPLICATION_ID)/revoke" \
 -X POST \
--H "Authorization: Bearer $(TOKEN)" \
+-H "Authorization: Bearer $(MANAGEMENT_API_TOKEN)" \
 -H "Content-Type: application/json" \
 -d "{\"token\":\"$(TOKEN_TO_REVOKE)\"}"'
 title="/revoke"
 />
 
+### Revoking token by token_id
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:121]:
```suggestion
### Revoking Token by ID
```

File: docs/workflows/api-token.md (hunk 3 above)
Comment (tomsik68) [reply docs/workflows/api-token.md:104]: Great call! I think I made this up, but in my mind, our APIs are divided to "management API" and "auth API" (`api-` vs `auth-` prefix).

In this context, "revoke a token via management API" simply means "revoke a token using something else than the RFC-7009 endpoint". Maybe it should be written as "In order to revoke a token by token_id" for consistency with section header. Would that make more sense?

Comment (turnipdabeets) [reply docs/workflows/api-token.md:104]: Ah ok, yea that might make more sense

File: docs/workflows/api-token.md [hunk 7]
```diff
@@ -95,6 +95,12 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
 
 ## Revoking Access Tokens
 
+Depending on what information is available to you, you may choose to use one of the ways to revoke an access token.
+
+If you have the raw access token, use the RFC-7009 revoke endpoint. If you only have the token ID (for example from active token listing), you can also revoke the token by token_id.
```
[the remark this answers is not in the document]
Comment (tomsik68) [reply docs/workflows/api-token.md:100]: The recommendation was wrong, let me explain more. Both endpoints achieve the same thing. There are two differences between them:

- RFC-7009 endpoint needs whole token, other endpoint only needs token_id
- RFC-7009 endpoint accepts bearer or basic auth, other endpoint only bearer

It really depends on the use case, let me give some examples.

1. Imagine an administrator configures an application to use referential tokens. Then, a user logs into the application and their token leaks somewhere. Now, the administrator wants to revoke the token. For this use case, it's only possible to use the RFC-7009 endpoint as the administrator has no way of knowing the token_id of the referential token.
2. Imagine a developer uses our management APIs to implement session management. They display a screen with a list of active tokens. For each token, they want to display a button that revokes the token. The token listing endpoint as specified provides only token_id. It doesn't provide the cryptographic string which makes it impossible to use the RFC-7009 endpoint.

[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:100]: Thanks, this makes a lot of sense!

Are they both `RFC-7009`? Or is that a typo?

[the remark this answers is not in the document]
Comment (tomsik68) [reply docs/workflows/api-token.md:100]: Sorry, that's a typo! Only one of them is RFC-7009 compliant.

File: docs/workflows/api-token.md [hunk 8]
```diff
@@ -95,6 +95,22 @@ In the Beyond Identity Admin Console under the "Apps" tab, select the "Beyond Id
 
 ## Revoking Access Tokens
 
+Beyond Identity exposes two different endpoints to revoke a token. Both endpoints achieve the same goal.
+
+There are two differences between them:
+
+- [RFC-7009](https://www.rfc-editor.org/rfc/rfc7009) endpoint needs whole token, management API only needs token_id
+- [RFC-7009](https://www.rfc-editor.org/rfc/rfc7009) endpoint accepts bearer or basic auth, management API accepts only bearer
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:103]: Should these both still be RFC-7009 if only one of them is compliant?

And then when you say endpoint needs/accepts ..., management API only needs ... you mean to make it compliant with  RFC-7009 it needs and accepts those but our endpoint only needs or accepts something different?

File: docs/workflows/api-token.md [hunk 9]
```diff
@@ -106,12 +122,29 @@ Note that passing an invalid token, or a token which has already been revoked or
...
+
+You must also know the `TOKEN_ID` of the token you want to revoke. This can be
+returned for example from the token listing.
+
+<MultiLanguageCodeBlock
+curl='curl "https://api-$(REGION).beyondidentity.com/v1/tenants/$(TENANT_ID)/realms/$(REALM_ID)/tokens/$(TOKEN_ID)" \
+-X DELETE \
+-H "Authorization: Bearer $(MANAGEMENT_API_TOKEN)" \
+-H "Content-Type: application/json"'
+title="DELETE /tokens/{token_id}"
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:145]:
```suggestion
title="DELETE /tokens/{TOKEN_ID}"
```

**Review by turnipdabeets — APPROVED** [review]
LGTM!

**Pull request merged**